src/ess/reduce/ui.py

```diff
@@ -232,3 +234,136 @@ def refresh_workflow_box(change) -> None:
     workflow_selection_box = widgets.HBox([workflow_select], layout=default_layout)
     workflow_box = widgets.Box(layout=default_layout)
     return widgets.VBox([workflow_selection_box, workflow_box])
+
+
+def _has_widget_value_setter(widget: widgets.Widget) -> bool:
+    widget_type = type(widget)
+    return (
+        widget_property := getattr(widget_type, 'value', None)
+    ) is not None and getattr(widget_property, 'fset', None) is not None
+
+
+def _get_parameter_box(widget: WorkflowWidget | ParameterBox) -> ParameterBox:
+    if isinstance(widget, WorkflowWidget):
+        return widget.parameter_box
+    elif isinstance(widget, ParameterBox):
+        return widget
+    else:
+        raise TypeError(
+            f"Expected target_widget to be a WorkflowWidget or ParameterBox, "
+            f"got {type(widget)}."
+        )
+
+
+def set_input_widget_values(
+    widget: WorkflowWidget | ParameterBox,
+    new_parameter_values: dict[str | type, dict[str, Any]],
+) -> None:
+    """Set the values of the input widgets in the target widget.
+
+    Example
+    -------
+    {
+        'WavelengthBins': {
+            'fields': {'start': 1.0, 'stop': 14.0, 'nbins': 500}
+        }
+    }
+
+    Parameters
+    ----------
+    widget:
+        The widget containing the input widgets.
+    new_parameter_values:
+        A dictionary of parameter values to set for the input widgets.
+        The keys are the parameter names or the type(Key in Sciline graphs).
+        The values are dictionaries with the following keys
+
+        - 'fields':
+            A dictionary of field names and values to set.
+            i.e. {'dim': 'tof', 'unit': 'mm'}
+            The field values will be set if the field exists in the widget.
+            Otherwise, the field value will be ignored and a warning will be raised.
+        - 'value':
+            If the widget has a value setter, the value will be set directly.
+            Otherwise, it will be ignored and a warning will be raised.
+
+    Raises
+    ------
+    TypeError:
+        If the target_widget is not a WorkflowWidget or a ParameterBox.
+
+    """
+    parameter_box = _get_parameter_box(widget)
+    # Walk through the existing input widgets and set the values
+    for node, widget in parameter_box._input_widgets.items():
+        new_internal_values = new_parameter_values.get(
+            node, new_parameter_values.get(parameter_box._input_registry[node].name, {})
+        )
```

Comment:
It first tries to find the `new_parameter_values` with a node(`type`) and if it can't find it it uses the `name` which is expected to be string version of type.

Reply:
My question is why you have chosen such a design.

Reply:
Because there are a few scenarios of making these `new_parameter_values` dictionaries.
1. Hard-coding as a dictionary:
  It'd be more convenient to just copy+paste the names from the parameter box in this case.
2. Use workflow inputs or default parameter dictionaries from modules.
  Users will also want to use the results of the workflow for setting parameters as well,
  which gives a dictionary with keys as `type`
3. Mix of both.

Reply:
Both 1.) and 2.) sound like it needs to be a type (Sciline key)? So should the string option be removed?

Remember it is always easier to add something later than to remove it. Any if you add support now you need to write (a) documentation and (b) more tests.

Reply:
> Both 1.) and 2.) sound like it needs to be a type (Sciline key)? So should the string option be removed?

I think that ruins the usability of this interface if we only allow types.
Then you need to know which module the types are from when you want to quickly update values from this interface.
So option 1) definitely needs string option...?

I think I'm confused. 
I thought the target users are people who don't want to use python interfaces directly and are using GUI instead.

If you can load the type for each target and everything, you won't need the GUI so much...?

Reply:
Conclusion from discussion: Remove the string option for now.